core/arch/arm/kernel/thread.c

```diff
@@ -167,24 +167,24 @@ void thread_check_canaries(void)
 	size_t n;
 
```

Comment:
It make more sense to have `panic_unless()` for the canaries than the old `assert()`.